src/Parquet.chpl

```diff
@@ -11,8 +13,34 @@ module Parquet {
   extern var ARROWINT64: c_int;
   extern var ARROWINT32: c_int;
   extern var ARROWUNDEFINED: c_int;
+  extern var ARROWERROR: c_int;
 
   enum ArrowTypes { int64, int32, notimplemented };
+
+  record parquetErrorMsg {
+    var errMsg: c_ptr(uint(8));
+    proc init() {
+      errMsg = c_nil;
+    }
+    
+    proc deinit() {
+      extern proc c_free_string(ptr);
+      c_free_string(errMsg);
+    }
+
+    proc parquetError(lineNumber, routineName, moduleName) throws {
+      extern proc strlen(a): int;
+      var err;
+      try! err = createStringWithNewBuffer(errMsg, strlen(errMsg));
```

Comment:
Is `try!` really what we want here?  We're already throwing, if we can't allocate a new buffer maybe we can use a default message?  I doubt this would ever really happen, but if it does, it takes out the server process.

Reply:
I believe that it is, it seems to be the convention used in other places in the Arkouda code base, but I could add an alternative default message in the failing case if you would prefer it that way

Reply:
We've been trying to remove the number of `try!` and `halts` and shift entirely to error handling.

Reply:
Alright, pushed that change up; is that something along the lines of what you had in mind?